app/routes/user/password.server.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+import { ChangePasswordCommand } from '@aws-sdk/client-cognito-identity-provider'
+
+import { getOpenIDClient, storage } from '../__auth/auth.server'
+import { getUser, parseTokenSet } from '../__auth/user.server'
+import { maybeThrow } from './cognito.server'
+import { cognito } from '~/lib/cognito.server'
+
+export module PasswordManager {
+  async function updatePasswordFunc(
+    request: Request,
+    oldPassword: string,
+    newPassword: string
+  ) {
+    const user = await getUser(request)
+    if (!user) throw new Response('not signed in', { status: 403 })
+    const client = await getOpenIDClient()
+    const session = await storage.getSession(request.headers.get('Cookie'))
+    const refreshToken = session.get('refreshToken')
+    const tokenSet = await client.refresh(refreshToken)
+    const parsedTokenSet = parseTokenSet(tokenSet)
```

Comment:
We have the access token stored in the session. It's not returned by the `getUser` function at the moment. I'll think about where we should expose it and get back to you on that.

Reply:
Sounds like a plan. Let me know what you come up with and I’ll add it in!